### PR TITLE
Four bar problem fix

### PR DESCRIPTION
**Pull request by GLRoylance**
## Overview

This PR deletes an unused variable (kills 1 VS reported problem). It adds JavaDoc comments to several methods. It adds explanation for RelativeEncoder scaling. It sets some temporary constant values (marked with TODO:).

## Design Doc

Relatively minor changes.

## Tests Ran

SimGUI

### CODE REVIEW

File: src/main/java/frc/robot/constants/ArmConstants.java [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 
-  // pid values - TBD
-  public static final double kP = -1;
-  public static final double kI = -1;
-  public static final double kD = -1;
-  public static final double kTolerance = -1;
-  public static final double minMotorPower = -1;
-  public static final double maxMotorPower = -1;
+  // TODO: PID values - TBD (fake values for now)
+  public static final double kP = 1.0;
```
Comment (AnthonyFurman) [inline src/main/java/frc/robot/constants/ArmConstants.java:8]: I think using a P value of zero, starting out, would be safer.

File: src/main/java/frc/robot/subsystems/FourBarArm.java [hunk 2]
```diff
@@ -14,32 +14,57 @@ public class FourBarArm extends SubsystemBase {
...
   @Override
   public void periodic() {
-    m_motor.set(MathUtil.clamp(m_pid.calculate(m_encoder.getPosition()), ArmConstants.minMotorPower, ArmConstants.maxMotorPower));
+    // calculate the PID power level
+    double pidPower = m_pid.calculate(m_encoder.getPosition());
+    // calculate the feedforward power (nothing for now)
+    double feedforwardPower = 0.0;
+
+    // set the motor power
+    m_motor.set(MathUtil.clamp(pidPower + feedforwardPower, ArmConstants.minMotorPower, ArmConstants.maxMotorPower));
```
Comment (AnthonyFurman) [inline src/main/java/frc/robot/subsystems/FourBarArm.java:61]: I don't think this is necessary. Yichen is working on the four bar arm feedforward. Please talk to us if you want to suggest changes like this.

File: src/main/java/frc/robot/constants/ArmConstants.java (hunk 1 above)
Comment (GLRoylance) [reply src/main/java/frc/robot/constants/ArmConstants.java:8]: With PID = (0, 0, 0), a simulation would never converge.

File: src/main/java/frc/robot/constants/ArmConstants.java [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
+  public static final double kI = 0.0;
+  public static final double kD = 0.0;
+  public static final double kTolerance = 0.02;
+  public static final double minMotorPower = -1.0;
+  public static final double maxMotorPower = +1.0;
```
Comment (AnthonyFurman) [inline src/main/java/frc/robot/constants/ArmConstants.java:13]: for safety I would use lower values

File: src/main/java/frc/robot/controls/Operator.java [hunk 4]
```diff
@@ -15,7 +15,7 @@ public class Operator {
    */
   public static void configureControls(FourBarArm arm) {
     operator.get(Button.Y).onTrue(new ExtendToPosition(arm, ArmConstants.topPosition));
-    operator.get(Button.X).onTrue(new ExtendToPosition(arm, ArmConstants.middlePosiiton));
+    operator.get(Button.X).onTrue(new ExtendToPosition(arm, ArmConstants.middlePosition));
     operator.get(Button.A).onTrue(new ExtendToPosition(arm, ArmConstants.intakePosition));
     operator.get(Button.Y).onTrue(new ExtendToPosition(arm, ArmConstants.shelfPosition));
```
Comment (AnthonyFurman) [inline src/main/java/frc/robot/controls/Operator.java:20]: @MagicEmu I know you were made aware of the Y button issue

File: src/main/java/frc/robot/constants/ArmConstants.java (hunk 1 above)
Comment (AnthonyFurman) [reply src/main/java/frc/robot/constants/ArmConstants.java:8]: Yes, simulation is working on a different branch.

**Comment by AnthonyFurman** [comment]
I would prefer @MagicEmu make these changes.
In the future, @GLRoylance , this feedback would be useful on the PR.

File: src/main/java/frc/robot/subsystems/FourBarArm.java (hunk 2 above)
Comment (GLRoylance) [reply src/main/java/frc/robot/subsystems/FourBarArm.java:61]: Bare code is often difficult to understand. I've added comments that explain the research I did to understand the code.

For PIDF, WPILIB has bottled up the PID but not the F. Separate them to make the pieces clear.

File: src/main/java/frc/robot/constants/ArmConstants.java (hunk 1 above)
Comment (GLRoylance) [reply src/main/java/frc/robot/constants/ArmConstants.java:8]: Safety might be better served by limiting the power to a small value such as 0.05.

I've been looking at the simulation branch. I like that it is using both the physics and mech2d. I also like the close method.

File: src/main/java/frc/robot/constants/ArmConstants.java (hunk 3 above)
Comment (GLRoylance) [reply src/main/java/frc/robot/constants/ArmConstants.java]: min/max power reduced to 0.05.

Comment (GLRoylance) [reply src/main/java/frc/robot/constants/ArmConstants.java]: kTolerance of 0.02 radian is about 1 degree.

File: src/main/java/frc/robot/subsystems/FourBarArm.java (hunk 2 above)
Comment (AnthonyFurman) [reply src/main/java/frc/robot/subsystems/FourBarArm.java:61]: Yes, I don't think the comments are really necessary to explain that research. We have the WPI documentation, and I did some work to explain PID + FF to the students during last semester.

File: src/main/java/frc/robot/constants/ArmConstants.java (hunk 3 above)
Comment (AnthonyFurman) [reply src/main/java/frc/robot/constants/ArmConstants.java]: kTolerance is fine, mainly meant min/max reduced. Though that being said, to be more clear, you can do Units.degreesToRadians(1) which is easier to understand and emphasizes the units used.

**Review by AnthonyFurman — APPROVED** [review]

**Pull request merged**